llvm/lib/Analysis/ValueTracking.cpp

```diff
@@ -429,10 +429,14 @@ void llvm::computeKnownBitsFromRangeMetadata(const MDNode &Ranges,
     ConstantInt *Upper =
         mdconst::extract<ConstantInt>(Ranges.getOperand(2 * i + 1));
     ConstantRange Range(Lower->getValue(), Upper->getValue());
+    unsigned RangeBitWidth = Lower->getBitWidth();
 
     // The first CommonPrefixBits of all values in Range are equal.
     unsigned CommonPrefixBits =
         (Range.getUnsignedMax() ^ Range.getUnsignedMin()).countl_zero();
+    // BitWidth must equal RangeBitWidth.  Otherwise Mask will be set
+    // incorrectly.
+    assert(BitWidth == RangeBitWidth && "BitWidth must equal RangeBitWidth");
```

Comment:
This is just an assert and comment change. This can be a separate patch, you should also avoid the temporary RangeBitWidth which is unused in release builds